prefetch.bs

```diff
@@ -963,6 +975,74 @@ The following parameters are defined for the `prefetch` token:
+
+<div class="issue">CDDL snippets use the "text" type instead of "browsingContext.BrowsingContext" to allow indepedent programmatic processing of CDDL snippets. Currently, other modules cannot be referenced.</div>
+
+<h3 id="speculation-module">The speculation module</h3>
+
+The speculation module contains commands for managing the remote end behavior for prefetches, prerenders, and speculation rules.
+
+<h3 id="speculation-module-definitions">Definitions</h3>
+
+{^Remote end definition^} and {^local end definition^}
+
+<pre class="cddl" data-cddl-module="remote-cddl,local-cddl">
+speculation.PreloadingStatus = "pending" / "ready" / "success" /
+                   "failure" / "not-supported"
+</pre>
+
+The <code>speculation.PreloadingStatus</code> type represents the different types of preloading statuses. This status is shared by prefetches and prerenders.
+
+"<dfn for="speculation.PreloadingStatus" cddl-value>pending</dfn>" status is defined when a [=prefetch record=] |prefetchRecord| begins to [=start a referrer-initiated navigational prefetch=].
+
+"<dfn for="speculation.PreloadingStatus" cddl-value>ready</dfn>" status is defined when a [=prefetch record=] |prefetchRecord| is [=prefetch record/complete=].
+
+"<dfn for="speculation.PreloadingStatus" cddl-value>success</dfn>" status is defined when a [=prefetch record=] |prefetchRecord| is activated, such as from the user agent's navigation.
+
+"<dfn for="speculation.PreloadingStatus" cddl-value>failure</dfn>" status is defined when a [=prefetch record=] |prefetchRecord| has failed, such as when the |prefetchRecord| expires.
+
+"<dfn for="speculation.PreloadingStatus" cddl-value>not-supported</dfn>" status is defined when a [=prefetch record=] |prefetchRecord| is ineligble to [=start a referrer-initiated navigational prefetch=].
+
+<h3 id="speculation-module-events">Events</h3>
+
+<pre highlight="cddl" class="cddl" data-cddl-module="remote-cddl,local-cddl">
+SpeculationEvent = (
+  speculation.PrefetchStatusUpdated
+)
+</pre>
+
+<h4 id="speculation-prefetchstatusupdated-event">The speculation.prefetchStatusUpdated Event</h4>
+
+<pre highlight="cddl" class="cddl" data-cddl-module="local-cddl">
+speculation.PrefetchStatusUpdated = (
+   method: "speculation.prefetchStatusUpdated",
+   params: speculation.PrefetchStatusUpdatedParameters
+)
+
+speculation.PrefetchStatusUpdatedParameters = {
+   context: text,  
+   url: text,
+   status: speculation.PreloadingStatus
+}
+</pre>
+
+<div algorithm="remote end event trigger for speculation.prefetchStatusUpdated">
+To <dfn>trigger a prefetch status updated event</dfn> given a [=navigable=] |navigable|, |prefetchRecord|, and a {^speculation.PreloadingStatus^} |preloadingStatus|:
+
+1. Let |frameId| be |navigable|'s [=navigable/top-level traversable=]'s [=navigable id=].
+1. Let |url| be the |prefetchRecord|'s [=prefetch record/URL=].
+1. Let |status| be the |preloadingStatus|.
```

Comment:
Aliasing |preloadingStatus| to |status| doesn't seem useful.

Generally since |frameId| and |url| are only used once, I'm not sure declaring them as variables is that helpful. (Especially |url|.)